cli_tools/common/utils/logging/service/log_entry.go

```diff
@@ -171,27 +188,23 @@ type OutputInfo struct {
 	FailureMessageWithoutPrivacyInfo string `json:"failure_message_without_privacy_info,omitempty"`
 }
 
-func (l *Logger) updateParams(projectPointer *string) {
+func (l *Logger) updateParams(params map[string]string) {
 	l.mutex.Lock()
 	defer l.mutex.Unlock()
 
-	if projectPointer == nil {
+	l.updateProject(params)
+}
+
+func (l *Logger) updateProject(params map[string]string) {
+	project, ok := params[param.UpdatedParamProject]
+	if !ok {
 		return
 	}
 
-	project := *projectPointer
 	obfuscatedProject := Hash(project)
 
-	if l.Params.ImageImportParams != nil {
-		l.Params.ImageImportParams.CommonParams.Project = project
-		l.Params.ImageImportParams.CommonParams.ObfuscatedProject = obfuscatedProject
-	}
-	if l.Params.ImageExportParams != nil {
-		l.Params.ImageExportParams.CommonParams.Project = project
-		l.Params.ImageExportParams.CommonParams.ObfuscatedProject = obfuscatedProject
-	}
-	if l.Params.InstanceImportParams != nil {
-		l.Params.InstanceImportParams.CommonParams.Project = project
-		l.Params.InstanceImportParams.CommonParams.ObfuscatedProject = obfuscatedProject
+	if commonParams := l.Params.commonParams(); commonParams != nil {
+		commonParams.Project = project
+		commonParams.ObfuscatedProject = obfuscatedProject
```

Comment:
Having project mutable, and allowing another object to mutate its field, makes it difficult for me to reason about it. Here are some issues that come up in my mind:
 1. What code is responsible for calling this method?
 2. What happens if it doesn't get called in the the future?
 3. Is there an order of events that need to be maintained?
 4. What code reads this field?